File: DataStructures/Java/Queue.java
public class Queue {
	int front, rear, capacity;
	long array[];

	Queue(int size) {
		front = rear = 0;
		capacity = size;
		array = new long[capacity];
	}

	public void enqueue(int item) {
		if (capacity == rear) {
			System.out.println("\nQueue is full\n");
			return;
		} else {
			array[rear] = item;
			rear++;
		}
	}

	public void dequeue() {
		if (front == rear) {
			System.out.println("\nQueue is empty\n");
			return;
		} else {
			for (int i = 0; i < rear - 1; i++) {
				array[i] = array[i + 1];
			}
			if (rear < capacity)
				array[rear] = 0;
			rear--;
		}
	}

	public void display() {
		int i;
		if (front == rear) {
			System.out.println("Queue is Empty\n");
			return;
		}
		for (i = front; i < rear; i++) {
			System.out.println(array[i]);
		}
	}

	public static void main(String[] args) {
		Queue queue = new Queue(10);
		queue.enqueue(5);
		queue.enqueue(6);
		queue.enqueue(7);
		queue.display();
		queue.dequeue();
		queue.display();

	}

}

File: DataStructures/TypeScript/Queue.ts
export class Queue {
  private elements: number[];
  private capacity: number;
  private rearIndex: number;

  constructor(size: number) {
    this.capacity = size;
    this.rearIndex = -1;
    this.elements = new Array<number>(size);
  }

  // Get the size of the list
  getSize(): number {
    return Math.max(this.rearIndex + 1, 0);
  }

  // Check if the list is empty
  isEmpty(): boolean {
    return this.getSize() === 0;
  }

  // Add an element to the end of the queue
  enqueue(element: number) {
    if (this.getSize() === this.capacity)
      throw new Error("Cannot enqueue into a full queue");

    this.elements[++this.rearIndex] = element;
  }

  // Remove an element from the front of the queue
  dequeue(): number {
    if (this.isEmpty()) throw new Error("Cannot dequeue from an empty queue");
    const element = this.elements[0];

    // Move all elements to the left by one position
    for (let i = 0; i < this.rearIndex; i++) {
      this.elements[i] = this.elements[i + 1];
    }
    this.rearIndex--;

    return element;
  }

  // Get the element at the front of the queue
  peek(): number {
    if (this.isEmpty()) throw new Error("Cannot peek into an empty queue");
    return this.elements[0];
  }

  // Print the elements of the queue
  display() {
    if (this.isEmpty()) throw new Error("Cannot display an empty queue");

    for (let i = 0; i <= this.rearIndex; i++) {
      console.log(this.elements[i]);
    }
  }
}
